fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/ToolManager.java

```diff
@@ -0,0 +1,96 @@
+package net.fabricmc.fabric.api.tool.attribute.v1;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.ActionResult;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.util.TriState;
+import net.fabricmc.fabric.impl.tool.attribute.ToolManagerImpl;
+
+/**
+ * API facing part to register tool handlers and get information about how tools are handled.
+ */
+public final class ToolManager {
+	/**
+	 * Returns a event for the tag provided, creates a new event if it does not exist.
+	 *
+	 * @param tag the tag provided for the tool
+	 * @return the event callback.
+	 */
+	public static Event<ToolHandler> tag(Tag<Item> tag) {
+		return ToolManagerImpl.tag(tag);
+	}
+
+	/**
+	 * Returns a event used for every tag registered.
+	 *
+	 * @return the event callback.
+	 */
+	public static Event<ToolHandler> general() {
+		return ToolManagerImpl.general();
+	}
```

Comment:
I don't think those events  are too useful for standard uses, as such they should probably be moved down in this class and the comment should refer to DynamicAttributeTool as the standard way to implement the functionality. The same applies to the class comment.